consul/resource_consul_config_entry_test.go

```diff
@@ -0,0 +1,278 @@
+
+\* '' has invalid keys: ThisFieldDoesNotExists`),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_config_entry.foo", "name", "foo"),
+					resource.TestCheckResourceAttr("consul_config_entry.foo", "kind", "service-defaults"),
+					resource.TestCheckResourceAttr("consul_config_entry.foo", "config_json", "{\"MeshGateway\":{},\"Protocol\":\"https\"}"),
+				),
+			},
+			{
+				Config: testAccConsulConfigEntry_ProxyDefaults,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_config_entry.foo", "name", "global"),
+					resource.TestCheckResourceAttr("consul_config_entry.foo", "kind", "proxy-defaults"),
+					resource.TestCheckResourceAttr("consul_config_entry.foo", "config_json", "{\"Config\":{\"foo\":\"bar\"},\"MeshGateway\":{}}"),
+				),
+			},
+			{
+				Config: TestAccConsulConfigEntry_ServiceRouter,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_config_entry.service_router", "name", "web"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_router", "kind", "service-router"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_router", "config_json", "{\"Routes\":[{\"Destination\":{\"Service\":\"admin\"},\"Match\":{\"HTTP\":{\"PathPrefix\":\"/admin\"}}}]}"),
+				),
+			},
+			{
+				Config: TestAccConsulConfigEntry_ServiceSplitter,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_config_entry.service_splitter", "name", "web"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_splitter", "kind", "service-splitter"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_splitter", "config_json", "{\"Splits\":[{\"ServiceSubset\":\"v1\",\"Weight\":90},{\"ServiceSubset\":\"v2\",\"Weight\":10}]}"),
+				),
+			},
+			{
+				Config: TestAccConsulConfigEntry_ServiceResolver,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_config_entry.service_resolver", "name", "web"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_resolver", "kind", "service-resolver"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_resolver", "config_json", "{\"DefaultSubset\":\"v1\",\"Subsets\":{\"v1\":{\"Filter\":\"Service.Meta.version == v1\"},\"v2\":{\"Filter\":\"Service.Meta.version == v2\"}}}"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccConsulConfigEntry_Errors(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() {},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccConsulConfigEntry_ProxyDefaultsWrongName,
+				ExpectError: regexp.MustCompile("Provider produced inconsistent result after apply: When applying changes to consul_config_entry.foo, provider \"consul\" produced an unexpected new value for was present, but now absent."),
```

Comment:
This error message seems like it's missing some words. That might not be something we can fix but assuming this test passes does that mean we are not forming the error message properly somewhere?

Reply:
I think I will need to handle this case separately to give the user a better error message. I will do it in a new PR to not delay this one any longer.